packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart

```diff
@@ -385,10 +453,7 @@ class WebKitWebViewController extends PlatformWebViewController {
     covariant WebKitNavigationDelegate handler,
   ) {
     _currentNavigationDelegate = handler;
-    return Future.wait(<Future<void>>[
-      _webView.setUIDelegate(handler._uiDelegate),
```

Comment:
The `WKUIDelegate` should actually be set even without a `NavigationDelegate`, so this has been moved to the constructor.